src/view_neume.cpp

```diff
@@ -93,249 +97,176 @@ void View::DrawNc(DeviceContext *dc, LayerElement *element, Layer *layer, Staff
-                    lineCode = SMUFL_E9BA_chantLigaturaDesc3rd;
-                    break;
-                case -3:
-                    lineCode = SMUFL_E9BF_chantConnectingLineAsc4th;
-                    break;
-                default:
-                    lineCode = SMUFL_E9B7_chantEntryLineAsc5th;
-                    break;
-            }
-            DrawNc(dc, dynamic_cast<LayerElement *>(ncVector.at(0)), layer, staff, measure);
-
-            if (lineCode == SMUFL_E9B9_chantLigaturaDesc2nd || lineCode == SMUFL_E9BA_chantLigaturaDesc3rd) {
-                DrawNc(dc, dynamic_cast<LayerElement *>(ncVector.at(1)), layer, staff, measure, lineCode, -noteHeight / 4, 0);
-                DrawNc(dc, dynamic_cast<LayerElement *>(ncVector.at(3)), layer, staff, measure, SMUFL_E990_chantPunctum, noteWidth, 0);
+            if (pitchDifferences.at(1) == 1) {
+                if (position == 1) {
+                    params.at(0).fontNo = SMUFL_E9B9_chantLigaturaDesc2nd;
+                }
+                else if (position == 2) {
+                    params.at(0).fontNo = SMUFL_E9BD_chantConnectingLineAsc2nd;
+                }
             }
-            else {
-                DrawNc(dc, dynamic_cast<LayerElement *>(ncVector.at(1)), layer, staff, measure, SMUFL_E990_chantPunctum, noteHeight/4, 0);
-                DrawNc(dc, dynamic_cast<LayerElement *>(ncVector.at(2)), layer, staff, measure, lineCode);
-                DrawNc(dc, dynamic_cast<LayerElement *>(ncVector.at(2)), layer, staff, measure, SMUFL_E990_chantPunctum);
-                DrawNc(dc, dynamic_cast<LayerElement *>(ncVector.at(3)), layer, staff, measure);
+            else if (pitchDifferences.at(1) == 2) {
+                if (position == 1) {
+                    params.at(0).fontNo = SMUFL_E9BA_chantLigaturaDesc3rd;
+                }
+                else if (position == 2) {
+                    params.at(0).fontNo = SMUFL_E9BD_chantConnectingLineAsc2nd;
+                }
             }
             break;
         }
         case PRESSUS:
         {
-            // Two punctums followed by a punctum inclinatum
-            // No additional logic required
-            
-            DrawNc(dc, dynamic_cast<LayerElement *>(ncVector.at(0)), layer, staff, measure);
-            DrawNc(dc, dynamic_cast<LayerElement *>(ncVector.at(1)), layer, staff, measure);
-            DrawNc(dc, dynamic_cast<LayerElement *>(ncVector.at(2)), layer, staff, measure,
-                    SMUFL_E991_chantPunctumInclinatum);
+            params.at(0).fontNo = (position == 2) ? SMUFL_E991_chantPunctumInclinatum : SMUFL_E990_chantPunctum;
             break;
         }
-        default: DrawLayerChildren(dc, neume, layer, staff, measure);
+        default: break;
+    } 
+    // If the nc is supposed to be a virga and currently is being rendered as a punctum
+    // change it to a virga
+    if (nc->HasAttribute("diagonalright", "u") && params.at(0).fontNo == SMUFL_E990_chantPunctum) {

```

Comment:
I assume this will eventually be changed to a `GetDiagonalRight()` but OK for now.